src/users/users.service.ts

```diff
@@ -59,23 +59,27 @@ export class UsersService {
     const user = await this.usersRepository.findOneOrFail(id);
     const { firstName, lastName, gender } = updateUserDto;
 
-    this.usersRepository.assign(user, { firstName, lastName, gender });
+    this.usersRepository.assign(user, {
+      ...(firstName && { firstName }),
+      ...(lastName && { lastName }),
+      ...(gender && { gender }),
```

Comment:
Can't you something like this when destructuring the updateUserDto object:
```
const { teacherInput, studentInput, ...userData } = updateUserDto
```
And do this

```suggestion
      ...userData
```